app/workers/fast_track/disability_compensation_job.rb

```diff
@@ -24,34 +24,41 @@ def perform(form526_submission_id, full_name)
       client = Lighthouse::VeteransHealth::Client.new(get_icn(form526_submission))
 
       begin
-        send_fast_track_engineer_email_for_testing(form526_submission_id)
-
-        bp_readings = FastTrack::HypertensionObservationData.new(client.get_resource('observations')).transform
-        return if no_recent_bp_readings(bp_readings)
-
-        pdf = pdf(full_name, filtered_bp_readings(bp_readings),
-                  filtered_medications(client.get_resource('medications')))
+        return if bp_readings(client).blank?
 
+        pdf = pdf(full_name, bp_readings(client), medications(client))
         upload_pdf_and_attach_special_issue(form526_submission, pdf)
       rescue => e
         Rails.logger.error 'Disability Compensation Fast Track Job failing for form' \
                            "id:#{form526_submission.id}. With error message: #{e.message}" \
                            "with backtrace: #{e.backtrace}"
+        send_fast_track_engineer_email_for_testing(form526_submission_id, e.message, e.backtrace)
         raise
       end
     end
 
     private
 
-    def send_fast_track_engineer_email_for_testing(form526_submission_id)
+    def bp_readings(client)
+      @bp_readings ||= client.get_resource('observations')
+      @bp_readings.present? ? FastTrack::HypertensionObservationData.new(@bp_readings).transform : []
+    end
+
+    def medications(client)
+      @medications ||= client.get_resource('medications')
+      @medications.present? ? FastTrack::HypertensionMedicationRequestData.new(@medications).transform : []
```

Comment:
Is it intended to create a new `HypertensionMedicationRequestData` every time this method is called?

Reply:
Same as above. Let me know if you want to chat about it.